starknet_py/abi/v1/parser_transformer.py

```diff
@@ -0,0 +1,65 @@
+import os
+
+import lark
+from lark import Transformer, v_args
+
+from starknet_py.cairo.data_types import ArrayType, CairoType, FeltType, Option
+
+
+class ParserTransformer(Transformer):
+    """
+    Transforms the lark tree into a CairoTypes.
+    """
+
+    # pylint: disable=no-self-use
+
+    def __default__(self, data: str, children, meta):
+        raise TypeError(f"Unable to parse tree node of type {data}")
+
+    @v_args(inline=True)
+    def start(self, value):
+        return value
+
+    def type(self, value):
+        return value[0]
+
+    def type_felt(self, _value):
+        return FeltType()
+
+    def type_uint(self, _value):
+        return FeltType()
+
+    def type_unit(self, _value):
+        return None
+
+    def type_option(self, _value):
+        return Option()
+
+    def type_array(self, value):
+        return ArrayType(value[0])
+
+    def struct(self, tokens):
+        return "::".join(token for token in tokens if isinstance(token, str))
```

Comment:
Can you explain this?

Reply:
Okay, don't we only want to take the second value that is `"test"`, "MyStruct"` etc. and not the whole token?

Reply:
If we join a token it will be represented as a string (the second value) automatically